tests/devapps/MSI/HelperService/MSIHelperService/Helper/MSIHelper.cs

```diff
@@ -0,0 +1,370 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Net.WebSockets;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Web;
+
+namespace MSIHelperService.Helper
+{
+    internal static class MSIHelper
+    {
+        private static HttpClient s_httpClient = new HttpClient();
```

Comment:
nit: consider using IHttpClientFactory from ASP.NET Core https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0

Reply:
thanks for sharing this, this is very effective on a web app